homeassistant/bootstrap.py

```diff
@@ -87,9 +87,11 @@ def from_config_dict(config: Dict[str, Any],
                              log_no_color)
 
     core_config = config.get(core.DOMAIN, {})
+    has_api_password = bool((config.get('http') or {}).get('api_password'))
```

Comment:
If a user has just `http:` in their config, the key exists but is `None`. The `or` solves both if key does not exist or when it's `None`.